### ISSUE TITLE
GcrsToItrsRotationModel issues

### ISSUE THREAD

**Issue by Pugazhenthi-Sivasankar** [Tudat/tudat#415]
Hello everyone!! I am making use of the 'GcrsToItrsRotationModel' in the orbit determination and propagation of GRACE and PROBA satellites. When I had implemented the 'SimpleRotationModelSettings' I did not run into any compilation or run time errors. However, I got less accurate results. To improve the accuracy, I am trying to use the 'GcrsToItrsRotationModel'. When I implemented it, I got the following compilation errors:
[gcrs_error.txt](https://github.com/Tudat/tudat/files/2223332/gcrs_error.txt)
I have included the script file 'earthOrbiterBasicStateEstimation.cpp' in the following link:
https://github.com/Pugazhenthi-Sivasankar/Pugazh1
I used the Doxygen documentation to implement the 'GcrsToItrsRotationModel'. I don't know what I am missing and I am open to any suggestions. Thanks in advance.

**Comment by Pugazhenthi-Sivasankar** [comment]
I included all the pre-processor directives (#include..) and the namespace identifiers mentioned in 'createRotationModel.h'. Though it reduced the number of compilation errors, the building of the application failed. Here are the recent compilation errors:
[gcrs_error2.txt](https://github.com/Tudat/tudat/files/2223967/gcrs_error2.txt)
The new script file 'earthOrbiterBasicStateEstimation.cpp' is available in the following link:
https://github.com/Pugazhenthi-Sivasankar/Pugazh1

**Comment by DominicDirkx** [comment]
The error seems to be of a similar type, first error output:

C:\tudat\tudatBundle\tudatExampleApplications\satellitePropagatorExamples\SatellitePropagatorExamples\earthOrbiterBasicStateEstimation.cpp:80:74: error: 'GcrsToItrsRotationModelSettings' was not declared in this scope

should tell you what you need to know. In the future, first look at the error output yourself, and try to find teh cause. If you cannot find it, explain what you have tried and why it didn't work.

**Comment by Pugazhenthi-Sivasankar** [comment]
Thank you for pointing me in the right direction. I found out that the error, "'GcrsToItrsRotationModelSettings' was not declared in this scope" was due to the USE_SOFA variable. Though I have switched on the SOFA in the top CMakeLists.txt, there was a problem with USE_SOFA in the files 'createRotationModel.h' and 'createRotationModel.cpp'. I removed the 'if' conditions for the USE_SOFA variable in both files and then there were no errors while building. However, while running the code I got run-time errors due to inconsistencies in the orientation of the ephemeris of various bodies used in the simulation. I am in the process of resolving this inconsistencies.
